plugin/core/restore_lines.py

```diff
@@ -7,15 +7,16 @@ def __init__(self):
         self.saved_lines = []  # type: List[dict]
```

Comment:
Feel free to correct me, I don't know if I used the new API correctly :)

Here is my feedback:
I want to highlight the differences between the new api, and the code before it:

The code before it did this:
1. It copied the `row`, `text`, `cursor` position of each line, right before sending the request.
2. On selecting a completion item, it inserted back the text on each row. 
END RESULT: It brought the lines in a state like they were when the completions were requested.

The code in this commit does this.
1. It copies the `change_id`, `change_region`(the region passed to the `transform_region_from`),  `text` and `cursor` position for each line, right before sending the request.
2. On selecting a completion item, the transform_region_from returns a `transform_region`.  `transform_region` is used for removing and insert back the previous text for each line. 
```
 view.erase(edit, transform_region)
 view.insert(edit, transform_region.begin(), saved_line['text'])

// I could probably use view.replace instead of erase and insert
```
END RESULT: It brings the lines in a state like they were when the completions were requested.

You may have noticed that I did this a bit differently, than what John did in his gist.
https://gist.github.com/jskinner/4dfa6d86f848e1e1e583883507369b67#file-async_edits-py-L238-L244
```
    def fill_completions(self, edits):
        # The server has sent us a series of edits it wants to be made when
        # the completion command is run, however the user may have edited the
        # buffer since the request want sent. Adjust the response from the
        # server to account for these edits
        for c in self.changes_since_sent:
            edits = [(transform_region(r, c), text) for r, text in edits]
```
If I done the same thing as John did in his gist:
I would have to adjust the response from the server to account for all view edits that occurred,
and apply all TextChange-s to all completion items.
But that to me looks like a lot of work.

For me it looks easier to adjust the view state to the state it was when the request is send, 
than to adjust the response from the server to account all the possible edits that may have had occurred. 

Either way, I really want to hear the feedback from you all
And don't take me wrong with what I just said :)